bazel/aar_with_jni.bzl

```diff
@@ -30,10 +30,22 @@ load("@google_bazel_common//tools/maven:pom_file.bzl", "pom_file")
 # creates a few underlying libraries, because of this the classes.jar in
 # the aar we built was empty. This rule separately builds the underlying
 # kt.jar file, and replaces the aar's classes.jar with the kotlin jar
-def aar_with_jni(name, android_library, proguard_rules = "", archive_name = "", visibility = None, native_deps = []):
-    if not archive_name:
-        archive_name = name
+def aar_with_jni(name, android_library, manifest, archive_name, native_deps = [], proguard_rules = "", visibility = []):
+    manifest_name = name + "_android_manifest"
+    android_binary_name = name + "_bin"
+    jni_archive_name = archive_name + "_jni"
+    pom_name = name + "_pom"
+    cc_lib_name = name + "_jni_interface_lib"
```

Comment:
I think this implicitly creating a dependency on the name of an external rule.(?) I'd propose we simply pass the name of the target as a parameter.